### PR TITLE
DHFPROD-6306: Handle all tabbed content on step save

### PR DESCRIPTION
**Pull request by wooldridge**
### Description

The step (Load, Map, Match, Merge) tabbed dialog has been updated so that clicking Save under either tab saves changes under all tabs. The user is no longer prompted to save changes when switching tabs.

- Step saving tests have been moved up to the Steps parent component since that's where the saving is now handled rather than in the forms children.
- Similar for tests of the "Discard changes?" dialog on cancel, as that is also now handled exclusively in the Steps parent.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/e2e/cypress/integration/curation/load/defaultIngestion.spec.tsx [hunk 1]
```diff
@@ -56,6 +56,7 @@ describe("Default ingestion ", () => {
     loadPage.stepNameInput().should("be.disabled");
     loadPage.stepDescriptionInput().clear().type("UPDATE");
     loadPage.saveButton().click();
+    cy.waitForAsyncRequest();
```
Comment (brucean52) [inline marklogic-data-hub-central/ui/e2e/cypress/integration/curation/load/defaultIngestion.spec.tsx:59]: I still don't think this function does anything. Navigating through the UI can be done using selectors that are based off of UI state changes. The cypress-wait-until library is a better option.

Comment (wooldridge) [reply marklogic-data-hub-central/ui/e2e/cypress/integration/curation/load/defaultIngestion.spec.tsx]: Thanks, I removed the instances that I added and didn't see any negative impact.

**Comment by wooldridge** [comment]
Fixed a step-saving issue that was keeping saves from happening from the Advanced tab. Force-pushed and this should fix the failures that involve not finding JSON property names in mapped docs.

**Comment by wooldridge** [comment]
e2e failures, reverting to version with the waitForAsyncRequest() calls after step saves.

File: marklogic-data-hub-central/ui/e2e/cypress/integration/curation/load/defaultIngestion.spec.tsx (hunk 1 above)
Comment (wooldridge) [reply marklogic-data-hub-central/ui/e2e/cypress/integration/curation/load/defaultIngestion.spec.tsx:59]: I reverted on account of e2e failures on the CI/CD side. e2e tests now pass.

**Review by xnikhil08 — APPROVED** [review]

**Review by briantang — APPROVED** [review]
Looks great! Tests pass.

**Pull request merged**